client/client.go

```diff
@@ -108,6 +130,58 @@ func (c *LambdaClient) Gql(uri string, query string, variables map[string]interf
 	return &body.Data, nil
 }
 
+func (c *LambdaClient) Do(req *http.Request) (*http.Response, error) {
+	functionName, path, err := parseUri(req.URL.String())
+	if err != nil {
+		return nil, err
+	}
+
+	// Copy additional headers from the req struct into lambda request headers
+	// go http.Header type doesn't align with the lambda header type
+	// so we just take the first value of the request header
+	headers := c.buildHeaders()
+	for k, v := range req.Header {
+		if _, ok := headers[k]; !ok {
+			headers[k] = v[0]
+		}
+	}
+
+	body, err := ioutil.ReadAll(req.Body)
+
+	data, err := json.Marshal(payload{
+		Headers:               headers,
+		HttpMethod:            req.Method,
+		QueryStringParameters: map[string]string{},
+		Path:                  *path,
+		Body:                  string(body),
+	})
+
+	lambdaResponse, err := c.invoker.Invoke(req.Context(), &lambda.InvokeInput{
+		FunctionName: functionName,
+		Payload:      data,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	// attempt to convert lambda response into http Response
+	var respPayload responsePayload
+	err = json.Unmarshal(lambdaResponse.Payload, &respPayload)
+
+	if err != nil {
+		return nil, err
+	}
+
+	resp := http.Response{
+		Body:       ioutil.NopCloser(bytes.NewBuffer([]byte(respPayload.Body))),
```

Comment:
You can save allocating that byte slice
```suggestion
		Body:       ioutil.NopCloser(bytes.NewBufferString(respPayload.Body)),
```